beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -723,7 +723,27 @@ func (s *Server) PublishBlockV2(w http.ResponseWriter, r *http.Request) {
 		s.publishBlock(ctx, w, r, true)
 	}
 }
+func (s *Server) versionHeaderFromCurrentSlot() string {
+	// attempt to get the current fork information
+	ce := slots.ToEpoch(s.GenesisTimeFetcher.CurrentSlot())
+	if ce >= params.BeaconConfig().FuluForkEpoch {
+		return version.String(version.Fulu)
+	} else if ce >= params.BeaconConfig().ElectraForkEpoch {
+		return version.String(version.Electra)
+	} else if ce >= params.BeaconConfig().DenebForkEpoch {
+		return version.String(version.Deneb)
+	} else if ce >= params.BeaconConfig().CapellaForkEpoch {
+		return version.String(version.Capella)
+	} else if ce >= params.BeaconConfig().BellatrixForkEpoch {
+		return version.String(version.Bellatrix)
+	} else if ce >= params.BeaconConfig().AltairForkEpoch {
+		return version.String(version.Altair)
+	} else {
+		return version.String(version.Phase0)
+	}
+}
 
+// nolint:gocognit
```

Comment:
Please do not ignore the linter. This function should be refactored to reduce complexity rather than suppressing the linter. 

If you need to ignore, can you state a good cause for doing so?

Reply:
we should refactor it, I just wanted to get this pr out with the least amount of changes for the perceived urgency, we should refactor this to clean it up but I was afraid of introducing more changes with that. I need to make sure the errors are being returned correctly and exiting the function if wrapped into new functions. 

if you feel I must handle it in this PR then I'll shift focus to this .

Reply:
If you want to follow up immediately after this PR, that works for me. I'm just cautious of people ignoring the linter and never fixing it